fmf/base.py

```diff
@@ -172,10 +172,29 @@ def _initialize(self, path):
 
     def _merge_plus(self, data, key, value, prepend=False):
         """ Handle extending attributes using the '+' suffix """
+
         # Nothing to do if key not in parent
         if key not in data:
             data[key] = value
             return
+
+        # Special handling for merging lists with dictionary
+        if isinstance(data[key], list) and isinstance(value, dict):
+            for list_item in data[key]:
+                if not isinstance(list_item, dict):
+                    "MergeError: Item '{0}' in {1} must be a dictionary.".format(
+                        list_item, self.name)
+                self._merge_special(list_item, value)
+            return
+        if isinstance(data[key], dict) and isinstance(value, list):
+            for list_item in value:
+                if not isinstance(list_item, dict):
+                    "MergeError: Item '{0}' in {1} must be a dictionary.".format(
+                        list_item, self.name)
+                self._merge_special(list_item, data[key])
```

Comment:
Btw, what's happening with those original `something+` attributes? I mean, we do the update of the respective `something` attribute but what happens with `something+`. Does `fmf` care about it? Does it matter if we modify it? I am asking because this code is in fact modifying the `something+` element as well.

Reply:
The `_merge_special()` method should take target data as the first argument and the second one as the source of the modification. Here, I believe, we need to create several copies of the parent dictionary and update them with the respective list items. Should be addressed in e2b9c19. @kkaarreell, please, have a look if everything's ok.

Reply:
I have done in the other way around. I have modified list items with dict instead and later replaced the original dict with the newly updated list. But I think that the `deepcopy` approach is better.